### ISSUE TITLE
[Bug]: react-table: table cell focus outline should disappear when Escape key is press

### ISSUE THREAD

**Issue by bsunderhus** [microsoft/fluentui#26323]
### Library

React Components / v9 (@fluentui/react-components)

### Are you reporting Accessibility issue?

None

### Reproduction

https://react.fluentui.dev/?path=/docs/preview-components-table--default#cell-navigation

## Actual Behavior

If you focus on a `TableCell` and press Escape the focus outline remains on the cell.

## Expected Behavior

Focus outline should be removed whenever Escape key is press.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by ling1726** [comment]
This is no longer a bug. The work in #26124 shows that Escape should actually _**trigger**_ the focus outline
